### ISSUE TITLE
Token parse error on {-# UNPACK #-}

### ISSUE THREAD

**Issue by Vlix** [haskell/haskell-language-server#1258]
### Your environment
Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:
```sh
haskell-language-server version: 0.8.0.0 (GHC: 8.6.5) (PATH: /home/flix/.config/Code/User/globalStorage/haskell.haskell/haskell-language-server-0.8.0-linux-8.6.5) (GIT hash: eb58f13f7b8e4f9bc771af30ff9fd82dc4309ff5)
Tool versions found on the $PATH
cabal:		2.4.1.0
stack:		2.5.1
ghc:		8.0.2
```
Which lsp-client do you use: VSCode
Describe your project (alternative: link to the project): https://github.com/yesodweb/wai/blob/master/warp/Network/Wai/Handler/Warp/Request.hs#L212
Contents of `hie.yaml`: none

### Expected behaviour
Should just parse the .hs file.

### Actual behaviour
HLS fails to process the file. (`Request.hs`, see link to project)

### Include debug information

<details>
<summary>
LSP logs:
</summary>

```
haskell-language-server-0.8.0-linux-8.6.5: Tokens.next failed to parse     {-# UNPACK #-} !Int -- running total byte count (excluding current header chunk) Parser did not consume entire stream, left: "!Int -- running total byte count (excluding current header chunk)"
CallStack (from HasCallStack):
  error, called at src/Ide/Plugin/Eval/Parse/Token.hs:134:19 in hls-eval-plugin-0.1.0.0-inplace:Ide.Plugin.Eval.Parse.Token
```
</details>

**Comment by Vlix** [comment]
Might be related to this one?
https://github.com/haskell/haskell-language-server/issues/1160#issuecomment-756709959

**Comment by jneira** [comment]
I think the problem is related with parser errors coming from the eval plugin (take look to latest [issues labeled](https://github.com/haskell/haskell-language-server/issues?q=is%3Aissue+is%3Aopen+label%3A%22component%3A+eval+plugin%22) with) 
And hopefully it will be fixed by https://github.com/haskell/haskell-language-server/pull/1232

**Comment by konn** [comment]
I confirmed manually that this bug will be solved #1232. I've just added the following test case to regression test:

```hs
{- | Won't panic on UNPACKs -}
module TUNPACK (THStatus(..)) where

type ByteString = String
type BSEndo = ByteString -> ByteString
type BSEndoList = [ByteString] -> [ByteString]

data THStatus = THStatus
    {-# UNPACK #-} !Int -- running total byte count
    BSEndoList -- previously parsed lines
    BSEndo -- bytestrings to be prepended

-- >>> "Yay! UNPACK pragma didn't do bad things!"
```
